src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,37 +1,35 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
 import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
 import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 public class MachineServiceImpl implements MachineService<Machine> {
-    private final BulldozerProducer bulldozerProducer;
-    private final ExcavatorProducer excavatorProducer;
-    private final TruckProducer truckProducer;
+    private final Map<Class<? extends Machine>, MachineProducer<? extends Machine>>
+            producers = new HashMap<>();
 
     public MachineServiceImpl() {
-        bulldozerProducer = new BulldozerProducer();
-        excavatorProducer = new ExcavatorProducer();
-        truckProducer = new TruckProducer();
+        producers.put(Bulldozer.class, new BulldozerProducer());
+        producers.put(Excavator.class, new ExcavatorProducer());
+        producers.put(Truck.class, new TruckProducer());
```

Comment:
we should not have hardcode or any logic in the constructor, initialize it in the field with `Map.of()` as a constant, or inject the map in the constructor from outside of this method